### PR TITLE
LIME-147 - Updating error handling

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

Oauth error handling

### Why did it change

To take in error code and description properly, and return access denied oauth error for integration with CRI callback to ipv core (e.g when a user escapes there journey in driving licence cri)

### CODE REVIEW

**Review by MMcC-GDS — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "di-ipv-cri-common-express",
-  "version": "0.0.29",
+  "version": "0.0.30",
```
Comment (merlinc) [inline package.json:3]: Don't update the version number in a PR - make a separate one with just this update, and then release via Github.

Otherwise, this can cause other PRs to accidentally be included in a release.

File: src/lib/error-handling.js [hunk 2]
```diff
@@ -13,10 +13,11 @@ module.exports = {
     let redirect_uri = req.session.authParams.redirect_uri;
 
     if (err.isAxiosError) {
-      const oauthError = err?.response?.data?.oauth_error;
+      const oauthError = err?.response?.data;
```
Comment (merlinc) [inline src/lib/error-handling.js:16]: Have you run this locally on the Fraud / KBV repos using yarn link? 

I'd expect that changing the properties of what is considered an error would break the tests

**Review by merlinc — CHANGES_REQUESTED** [review]

File: src/lib/error-handling.js (hunk 2 above)
Comment (merlinc) [reply src/lib/error-handling.js:16]: e.g. https://github.com/alphagov/di-ipv-cri-kbv-front/blob/main/test/mocks/mappings/answer-error.json

File: src/lib/error-handling.test.js [hunk 3]
```diff
@@ -82,7 +80,7 @@ describe("error-handling", () => {
       });
 
       it("should use a default for error code", async () => {
```
Comment (merlinc) [inline src/lib/error-handling.test.js:82]: You're adding a new property, `message` to the error middleware.

Instead of changing the existing test, can you add a new test for the functionality you're adding. If you care about the order of resolution, you might want to have some extra tests as well.

File: .prettierrc.json [hunk 4]
```diff
@@ -1 +0,0 @@
-{}
```
Comment (merlinc) [inline .prettierrc.json:1]: Is this change relevant to LIME-147?

**Review by merlinc — CHANGES_REQUESTED** [review]
